test/optimus/pr/ValidateOptimus.wdl

```diff
@@ -1,64 +1,223 @@
+         input:
+            bam = bam,
+            expected_checksum = expected_bam_hash
+     }
 
-    # catch intermittent failures
-    set -eo pipefail
+     call ValidateMatrix as ValidateMatrix {
+         input:
+             matrix = matrix,
+             matrix_row_index = matrix_row_index,
+	     matrix_col_index = matrix_col_index,
+	     reference_matrix = reference_matrix
+     }
+
+     call ValidateMetrics {
+         input:
+             cell_metrics = cell_metrics,
+             gene_metrics = gene_metrics,
+             expected_cell_metric_hash = expected_cell_metric_hash,
+             expected_gene_metric_hash = expected_gene_metric_hash
+     }
 
-    # calculate hashes; awk is used to extract the hash from the md5sum output that contains both
-    # a hash and the filename that was passed. gzipped files are unzipped to avoid hashing malleable
-    # metadata
+     call GenerateReport as GenerateReport {
+         input:
+             bam_validation_result = ValidateBam.result,
+             matrix_validation_result = ValidateMatrix.result,
+             metric_and_index_validation_result = ValidateMetrics.result
+    }
+}
 
+task ValidateBam {
+    File bam
+    String expected_checksum
+    Int required_disk = ceil(size(bam, "G") * 1.1)
 
-    unzip "${matrix}"
-    matrix_hash=$(find . -name "*.npy" -type f -exec md5sum {} \; | sort -k 2 | md5sum | awk '{print $1}')
-    gene_metric_hash=$(zcat "${gene_metrics}" | md5sum | awk '{print $1}')
-    cell_metric_hash=$(zcat "${cell_metrics}" | md5sum | awk '{print $1}')
+    command <<<
+        echo Starting checksum generation...
+        # calculate hash for alignment positions only (a reduced bam hash)
+        samtools view -F 256 "${bam}" | cut -f 1-11 | md5sum | awk '{print $1}' > md5_checksum_reduced.txt
+        echo Reduced checksum generation complete
 
-    # calculate hash as above, but ignore run-specific bam headers
-    bam_hash=$(samtools view "${bam}" | md5sum | awk '{print $1}')
+        calculated_checksum=$(cat md5_checksum_reduced.txt)
 
-    # test each output for equivalence, echoing any failure states to stdout
-    fail=false
-    if [ "$bam_hash" != "${expected_bam_hash}" ]; then
-      >&2 echo "bam_hash ($bam_hash) did not match expected hash (${expected_bam_hash})"
-      fail=true
-    fi
+        if [ "$calculated_checksum" == ${expected_checksum} ]
+        then
+             echo Computed and expected bam hashes match \( $calculated_checksum \)
+             printf PASS > result.txt
+        else 
+             echo Computed \( $calculated_checksum \) and expected \( ${expected_checksum} \) bam file hashes do not match
+             printf FAIL > result.txt
+        fi
+    >>>
+  
+    runtime {
+        docker: "quay.io/humancellatlas/secondary-analysis-samtools:v0.2.2-1.6"
+        cpu: 1
+        memory: "3.75 GB"
+        disks: "local-disk ${required_disk} HDD"
+    }
+
+    output {
+        String result = read_string("result.txt")
```

Comment:
Did you consider `Boolean result = read_boolean("result.txt")`?

Reply:
Yes but I get a bit confused with true in bash due to inversion of regular 1/0 scemantics and I wasn't sure what Boolean was here so avoided it. Also thought String can potentially pass other statuses, e.g. warning better. Happy to change but will need a bit of help